pkg/controller/csr/csr_controller.go

```diff
@@ -222,8 +231,27 @@ func (r *reconcileCSR) Reconcile(ctx context.Context, request reconcile.Request)
 		}
 		reqLogger.V(5).Info("Signed CSR with name : %v.", csr.Name)
 	}
+	needsCSRRole := instance.Spec.CertificateManagement != nil
+	if !needsCSRRole && r.enterpriseCRDExists {
+		monitorCR := &operatorv1.Monitor{}
+		if err = r.client.Get(ctx, utils.DefaultTSEEInstanceKey, monitorCR); err != nil {
+			if apierrors.IsNotFound(err) {
+				return reconcile.Result{}, nil
+			}
+			return reconcile.Result{}, err
+		}
+		needsCSRRole = monitorCR.Spec.ExternalPrometheus != nil
+	}
 
-	return reconcile.Result{}, nil
+	var passthrough render.Component
+	if needsCSRRole {
+		// This controller creates the cluster role for any pod in the cluster that requires certificate management.
+		passthrough = render.NewPassthrough(certificatemanagement.CSRClusterRole())
+	} else {
+		passthrough = render.NewDeletionPassthrough(certificatemanagement.CSRClusterRole())
+	}
+	componentHandler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
```

Comment:
I think we probably want to create this prior to the loop above?

It's a bit of a niche scenario, but couldn't an error in the above loop result in us not creating / updating / deleting this, resulting in a deadlock? 

I think in practice it's unlikely because we skip CSRs that aren't for us, but if there's a CSR that _is_ for us somehow and is causing problems in the loop above, then we'll never execute this code until that CSR is resolved. Maybe that's OK, but I like to avoid unnecessary dependencies like this because they introduce complexity into the debugging process. It seems a more reasonable ordering of dependencies that we'd never handle any CSRs until the ClusterRole that gives our components permissions to create CSRs had been created / updated.

Reply:
Also, I think for now `needsCSRRole` is the same as "Should this controller do anything?" so we can probably short-circuit the controller if we move this earlier too?